File: Python/py_w1d3/hackathon/ninjas_vs_pirates_extra/classes/ninja.py
import random

class Ninja:
    def __init__(self, name):
        self.name = name
        self.strength = 10
        self.speed = 5
        self.health = 100
        self.damage = random.randint(1, 40)
        self.heal = 0

    def show_stats(self):
        print(
            f"Name: {self.name}\nStrength: {self.strength}\nSpeed: {self.speed}\nHealth: {self.health}\n"
        )

    def attack(self, enemy, tool):
        print(
            f"[ATTACK]{self.name} attacked with a {tool} and gave {enemy.name} -{self.damage+self.strength}pts"
        )
        enemy.health -= self.strength + self.damage
        return self

    def healing(self, food):
        print(food)
        if food == "apple":
            self.health += self.heal + 15
        if food == "herb":
            self.health += self.heal + 30
        print(f"{self.name} used {food} and increased {self.healing}pts")

File: MERN/MERN_w1/w1d2/ninja.js
class Ninja {
    constructor()
    {
        this.name = "Ryu"
        this.health = 100;
        this.speed = 3;
        this.strength = 3;
    }

    sayName()
    {
        console.log(this.name)
    }

    showStats()
    {
        console.log(this.health, this.speed, this.strength)
    }

    drinkSake()
    {
        this.health += 10;
        console.log(this.health);
    }
}

const newNinja = new Ninja;

newNinja.sayName();
newNinja.showStats();
newNinja.drinkSake();
